### PR TITLE
Add table acceleration flyout

### LINKED ISSUE #127: [FEATURE] Table Acceleration Support
Flint Spark extension to create skipping index with OpenSearch which can accelerate query with applicable filtering condition. 

### Skipping Index 
* User can only create, view or delete  a skipping index 
* One table can only have 1 skipping index for a Flint table
* Name is auto-generated for a given table and follows a pre-defined by a naming convention
* Columns can be accelerated individually using partition, value_set and min_max skipping type
* This is always auto-refreshed or manually refreshed. 
### Covering Index
* User can only create, view or delete a covering index 
* User can have multiple covering index for a Flint table 
* User can define their own name for the covering Index
* User can select columns required for creating the covering index
* This is always auto-refreshed or manually refreshed. 
### Materialized View
* User can only create, view or delete a materialised view 
* User can have multiple materialized views for a Flint table 
* User can define their own name for the materialized view
* Columns can be aggregated into metrics with different functions like count, min, max, sum and avg.  
* The above metrics can be further grouped into different columns [Ex. TUMBLE() or SLIDE()]
* Final check for MVs are where clauses to restrict metrics on a given subset of data
* This is always auto-refreshed or manually refreshed.

### PR DESCRIPTION
**Pull request by ps48**
### Description
Introduces acceleration index creation flyout in workbench
NOTE: This code is still in high flux, will add tests in a follow up PR. 
 
### Issues Resolved
#127 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Swiddis — COMMENTED** [review]
Some initial comments:
- It would be nice to have a screenshot or video in the PR description so we can see how it looks without having to start it up ourselves.
- I think that as a team we're hoping to be better about our testing; having some tests available for this would be helpful.

I'll look more carefully at the code when I have more time.

**Comment by ps48** [comment]
@Swiddis The UI/UX is changing quickly the current code will change soon (early next week). In my opinion, any test work now will be throw away. Also, Here's a short demo:


https://github.com/opensearch-project/dashboards-query-workbench/assets/4348487/401443c6-a4cf-4e2c-afaf-80d0f83c45c2

**Review by joshuali925 — COMMENTED** [review]
not familiar with acceleration details, left some general comments

File: public/components/acceleration/visual_editors/index_setting_options.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    },
+    {
+      id: `${idPrefix}1`,
+      label: 'Refresh by interval',
+    },
+  ];
+  const [indexName, setIndexName] = useState('');
+  const [indexAlias, setIndexAlias] = useState('');
+  const [primaryShards, setPrimaryShards] = useState('5');
+  const [replicaCount, setReplicaCount] = useState('1');
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/acceleration/visual_editors/index_setting_options.tsx]: `e.target.value` always returns a string. Have to handle explicit conversion either way. Converted to number in useState.

File: public/components/acceleration/visual_editors/index_setting_options.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  return (
+    <>
+      <EuiFormRow
+        label="Index name"
+        helpText='Must be in lowercase letters. Cannot begin with underscores or hyphens. Spaces, commas, and characters :, ", *, +, /, , |, ?, #, > are not allowed. Skipping Index names are pre-generated and cannot be changed.'
+      >
+        <EuiFieldText
+          placeholder="Enter Index Name"
+          value={indexName}
+          onChange={(e) => onChangeIndexName(e)}
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/acceleration/visual_editors/index_setting_options.tsx]: Thanks updated: https://github.com/opensearch-project/dashboards-query-workbench/pull/128/commits/046d3e5747b60a3f71fa34c2eb87aeeec47235dd

File: public/components/acceleration/visual_editors/skipping_index/add_fields_modal.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'dataType',
+      name: 'Datatype',
+      sortable: true,
+      truncateText: true,
+    },
+  ] as Array<EuiTableFieldDataColumnType<DataTableFieldsType>>;
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/acceleration/visual_editors/skipping_index/add_fields_modal.tsx]: Updated: https://github.com/opensearch-project/dashboards-query-workbench/pull/128/commits/046d3e5747b60a3f71fa34c2eb87aeeec47235dd

File: public/components/acceleration/visual_editors/skipping_index/add_fields_modal.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+          onClick={() => {
+            setAccelerationFormData({
+              ...accelerationFormData,
+              skippingIndexQueryData: [
+                ...accelerationFormData.skippingIndexQueryData,
+                ...selectedFields.map((x) => {
+                  return { ...x, accelerationMethod: 'PARTITION' };
+                }),
+              ],
+            });
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/acceleration/visual_editors/skipping_index/add_fields_modal.tsx:92]: I'll do changes for this in the next PR.

File: public/components/acceleration/visual_editors/covering_index/covering_index_builder.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            const columnValue = columns + `${n.label}`;
+            if (index !== selectedOptions.length - 1) return `${columnValue}, `;
+            else return columnValue;
+          },
+          ''
+        ) +
+        ')';
+    }
+    setColumnsValue(expresseionValue);
+  }, [selectedOptions]);
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/acceleration/visual_editors/covering_index/covering_index_builder.tsx]: updated: https://github.com/opensearch-project/dashboards-query-workbench/pull/128/commits/046d3e5747b60a3f71fa34c2eb87aeeec47235dd

File: public/components/acceleration/visual_editors/query_builder.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  return _.reduce(
+    skippingIndexQueryData,
+    function (columns, n, index) {
+      const columnValue = columns + `   ${n.fieldName} ${n.accelerationMethod}`;
+      if (index !== skippingIndexQueryData.length - 1) return `${columnValue}, \n`;
+      else return `${columnValue} \n`;
+    },
+    ''
+  );
+};
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/acceleration/visual_editors/query_builder.tsx:19]: Not really, all the options here are coming from combo box which we provide and the final query is copied to the editor. The backend might have to handle injection cases. But from UI side we provide empty query bar for now.

File: opensearch_dashboards.json [hunk 7]
```diff
@@ -1,11 +1,11 @@
 {
   "id": "queryWorkbenchDashboards",
-  "version": "3.0.0.0",
-  "opensearchDashboardsVersion": "3.0.0",
```
Comment (joshuali925) [inline opensearch_dashboards.json:4]: revert this change? main should be 3.0.0

File: package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "opensearch-query-workbench",
-  "version": "3.0.0.0",
```
Comment (joshuali925) [inline package.json:3]: same here

File: public/components/acceleration/visual_editors/skipping_index/delete_fields_modal.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+      sortable: true,
+      truncateText: true,
+    },
+    {
+      field: 'accelerationMethod',
+      name: 'Acceleration method',
+      sortable: true,
+      truncateText: true,
+    },
+  ] as Array<EuiTableFieldDataColumnType<SkippingIndexRowType>>;
```
Comment (joshuali925) [inline public/components/acceleration/visual_editors/skipping_index/delete_fields_modal.tsx:52]: also this possible to avoid assertions like https://github.com/opensearch-project/dashboards-query-workbench/pull/128#discussion_r1335250877?

**Review by mengweieric — APPROVED** [review]

File: public/components/acceleration/visual_editors/skipping_index/delete_fields_modal.tsx (hunk 9 above)
Comment (ps48) [reply public/components/acceleration/visual_editors/skipping_index/delete_fields_modal.tsx]: Updated: https://github.com/opensearch-project/dashboards-query-workbench/pull/128/commits/b0f24f30380e3940f537c14bce27139664b71327

File: package.json (hunk 8 above)
Comment (ps48) [reply package.json]: Sorry missed this one earlier: https://github.com/opensearch-project/dashboards-query-workbench/pull/128/commits/b0f24f30380e3940f537c14bce27139664b71327

File: opensearch_dashboards.json (hunk 7 above)
Comment (ps48) [reply opensearch_dashboards.json:4]: reverted: https://github.com/opensearch-project/dashboards-query-workbench/pull/128/commits/b0f24f30380e3940f537c14bce27139664b71327

**Review by joshuali925 — APPROVED** [review]
